operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakController.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.operator.v2alpha1;
+
+import javax.inject.Inject;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.api.reconciler.*;
+import io.javaoperatorsdk.operator.api.reconciler.Constants;
+import org.jboss.logging.Logger;
+import org.keycloak.operator.v2alpha1.crds.Keycloak;
+import org.keycloak.operator.v2alpha1.crds.KeycloakStatus;
+
+@ControllerConfiguration(namespaces = Constants.WATCH_CURRENT_NAMESPACE, finalizerName = Constants.NO_FINALIZER)
+public class KeycloakController implements Reconciler<Keycloak> {
+
+    @Inject
+    Logger logger;
+
+    @Inject
+    KubernetesClient client;
+
+    @Override
+    public UpdateControl<Keycloak> reconcile(Keycloak kc, Context context) {
+        logger.trace("Reconcile loop started");
+        final var spec = kc.getSpec();
+
+        logger.info("Reconciling Keycloak: " + kc.getMetadata().getName() + " in namespace: " + kc.getMetadata().getNamespace());
+
+        KeycloakStatus status = kc.getStatus();
+        var deployment = new KeycloakDeployment(client);
+
+        try {
+            var kcDeployment = deployment.getKeycloakDeployment(kc);
+
+            if (kcDeployment == null) {
```

Comment:
Not sure about the target scope of this operator, looks great as a baseline. 
Just maybe a small comment here. The situation might be covered when, the target deployment specs is different from the one is already there. In that case you might want to update the deployment. This is to cover the situation when 3rd party (accidentally or intentionally) changed the deployment. 
Also it's a good practice to add an Informer for the deployment, currently if the deployment is created, the reconciliation won't be triggered if the deployment changes.

Hope it helps :)

Reply:
You are right, this business logic is mostly "dummy" to keep the baseline as simple as possible, but we should move to shared informers as a next step.
Thanks for the suggestions!